src/pkg/repository/manager.go

```diff
@@ -15,16 +15,26 @@
 package repository
 
 import (
+	"context"
 	"github.com/goharbor/harbor/src/chartserver"
-	"github.com/goharbor/harbor/src/common/dao"
+	common_dao "github.com/goharbor/harbor/src/common/dao"
 	"github.com/goharbor/harbor/src/common/models"
 	"github.com/goharbor/harbor/src/pkg/project"
+	"github.com/goharbor/harbor/src/pkg/repository/dao"
 )
 
+// Mgr is the global repository manager instance
+// TODO after the refactor, the params should be removed
+var Mgr = New(nil, nil)
+
+// TODO refactor the repository manager
+
 // Manager is used for repository management
 // currently, the interface only defines the methods needed for tag retention
 // will expand it when doing refactor
 type Manager interface {
+	// if the repository with the specific name exists, read it, or create it
+	GetOrCreate(ctx context.Context, repository *models.RepoRecord) (created bool, id int64, err error)
```

Comment:
Makes no sense to call it `GetOrCreate` here, it's the manager layer it should be called `Create` only?

Reply:
Updated